src/parser.rs

```diff
@@ -606,24 +613,23 @@ pub fn script_data_long_string(input: &[u8]) -> IResult<&[u8], &str> {
 }
 
 pub fn script_data_date(input: &[u8]) -> IResult<&[u8], ScriptDataDate> {
-  let (i, (date_time, local_date_time_offset)) = pair(be_f64, be_i16)(input)?;
-  let date = ScriptDataDate {
-    date_time,
-    local_date_time_offset,
-  };
-  Ok((i, date))
+  pair(be_f64, be_i16)(input).map(|(i, (date_time, local_date_time_offset))| {
+    (
+      i,
+      ScriptDataDate {
+        date_time,
+        local_date_time_offset,
+      },
+    )
+  })
 }
 
-pub fn script_data_ecma_array(input: &[u8]) -> IResult<&[u8], Vec<ScriptDataObject<'_>>> {
-  let (i, _) = be_u32(input)?;
-  script_data_objects(i)
+pub fn script_data_ecma_array(input: &[u8]) -> IResult<&[u8], Vec<ScriptDataObject>> {
+  be_u32(input).and_then(|(i, _)| script_data_objects(i))
 }
 
-pub fn script_data_strict_array(input: &[u8]) -> IResult<&[u8], Vec<ScriptDataValue<'_>>> {
-  match be_u32(input) {
-    Ok((i, o)) => many_m_n(1, o as usize, script_data_value)(i),
-    Err(err) => Err(err),
-  }
+pub fn script_data_strict_array(input: &[u8]) -> IResult<&[u8], Vec<ScriptDataValue>> {
+  be_u32(input).and_then(|(i, o)| many_m_n(1, o as usize, script_data_value)(i))
```

Comment:
And here

Reply:
I wasn't able to change this one. Could you help me @sdroege?

Reply:
Looking at the https://docs.rs/nom/latest/nom/multi/fn.many_m_n.html description, it repeats the embedded parser N times, failing if the embedded parser does not succeed at least 1 time in our case. So I think we can try `length_count` :)

Reply:
But `length_count` would fail if it's less than N items or not?

Reply:
Yes, it fails if the parser does not run *exactly* N times. So we don't get any `Vec` if it is `<=N`. We can't use `length_count` then.